File: 27-04/sinal_fat_quad.py
def fatorial(n):
    if n == 0:
        return 1
    elif n > 0:
        return n * (fatorial(n - 1))
    
def sinal(n):
    if n > 0:
        return 1
    elif n < 0:
        return -1
    elif n == 0:
        return 0

def funcSinalFatorialQuadrado(n):
    return sinal(fatorial(n * n))

if __name__ == "__main__":
    print("Digite um número para calcular o sinal do fatorial ao quadrado:")
    print(funcSinalFatorialQuadrado(int(input())))

File: 27-04/sinal_fat_quad.hs
sinal :: Int -> Int
sinal n
    | n > 0 = 1
    | n < 0 = -1
    | n == 0 = 0

fatorial :: Int -> Int
fatorial n 
    | n == 0 = 1
    | n > 0 = n * (fatorial (n - 1))

funcSinalFatorialQuadrado :: Int -> Int
funcSinalFatorialQuadrado n = sinal (fatorial (n * n))


main :: IO ()
main = do 
    putStrLn "Digite um número para calcular o sinal do fatorial de n ao quadrado:"
    n <- readLn
    putStrLn "O sinal do fatorial é:"
    print (funcSinalFatorialQuadrado n)